packages/web-api-scan-runner/src/runner/runner.ts

```diff
@@ -24,13 +25,24 @@ import { WebDriverTask } from '../tasks/web-driver-task';
 
 // tslint:disable: no-null-keyword no-any
 
-function convertAxeToHtml(results: AxeResults): string {
-    return 'html report';
+function convertAxeToHtml(results: AxeResults, pageTitle: string): string {
```

Comment:
let this be a class AxeResultToHtmlConverter & then let the class take reporterFactory via constructor & use dependency injection.

Reply:
```
@injectable()
export class AxeResultToHtmlConverter {
    constructor(@inject(iocTypes.ReporterFactory) private readonly repoertFactory: typeof reporterFactory) {}
    convert(results: AxeResults, pageTitle: string): string {}
}
```

Reply:
another refactoring could be, you could have a ReportGenerator class that is responsible of creating the two report types. & will return the report[] ({reportType: 'html'| 'sarif', reportId: string})